### PR TITLE
Fix USB errors

### PR DESCRIPTION
**Pull request by dzid26**
Panda needed more time to be detected by USB host after reset.

USB errors would would show for the devices with USB hubs.

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
A `time.sleep` is rarely the right fix. Why doesn't the polling below that line take care of this?

**Comment by dzid26** [comment]
I got rid of the extra delay

File: board/jungle/scripts/reset_test.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
```
Comment (adeebshihadeh) [inline board/jungle/scripts/reset_test.py:2]: this is a good test to evaluate this change, but I don't think it has value going forward. I'd remove it

File: python/__init__.py [hunk 2]
```diff
@@ -451,7 +452,7 @@ def reconnect(self):
     # wait up to 15 seconds
     for _ in range(15*10):
       try:
-        self.connect()
+        self.connect(False, True)
```
Comment (adeebshihadeh) [inline python/__init__.py:455]: use the kwargs
```suggestion
        self.connect(claim=False, wait=True)
```

File: python/__init__.py [hunk 3]
```diff
@@ -347,7 +347,7 @@ def spi_connect(cls, serial, ignore_version=False):
     return None, handle, spi_serial, bootstub, None
 
   @classmethod
-  def usb_connect(cls, serial, claim=True):
+  def usb_connect(cls, serial, claim=True, wait=False):
```
Comment (adeebshihadeh) [inline python/__init__.py:350]: call it `silent`? wait doesn't mean anything in the context of this function

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
LGTM aside from the comments

File: python/__init__.py [hunk 4]
```diff
@@ -357,7 +357,8 @@ def usb_connect(cls, serial, claim=True):
           try:
             this_serial = device.getSerialNumber()
           except Exception:
-            logger.exception("failed to get serial number of panda")
+            if not wait:
```
Comment (adeebshihadeh) [inline python/__init__.py:360]: we should leave a good comment about why this is here

File: python/__init__.py (hunk 3 above)
Comment (dzid26) [reply python/__init__.py]: I called it `no-error`

**Pull request merged**